### PR TITLE
PSP-5951 refactor left hand file navigation - Acquisition ONLY

### PR DESCRIPTION
**Pull request by asanchezr**
Left hand sidebar navigation refactored to use routing instead of internal state. Only applies to acquisition file forms (for now)

### File routing
image

### File Property routing
image

### CODE REVIEW

**Comment by devinleighsmith** [comment]
something in this change has appeared to cause a regression in the tab view: 
image

File: source/frontend/src/features/mapSideBar/acquisition/FilePropertyRouter.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+};
+
+const getAcquisitionFileProperty = (
+  acquisitionFile: Api_AcquisitionFile,
+  selectedMenuIndex: number,
+) => {
+  const properties = acquisitionFile?.fileProperties || [];
+  const selectedPropertyIndex = selectedMenuIndex - 1;
+  const acquisitionFileProperty = properties[selectedPropertyIndex];
+  if (!!acquisitionFileProperty.file) {
```
[the remark this answers is not in the document]
Comment (asanchezr) [reply source/frontend/src/features/mapSideBar/acquisition/FilePropertyRouter.tsx:94]: I see. Didn't consider that edge case. Will update the error message.

**Comment by devinleighsmith** [comment]
our default handling of when a file fails to load is not very good. Predates this change so I'll log tech debt separately.

File: source/frontend/src/features/mapSideBar/acquisition/AcquisitionRouter.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+export const AcquisitionRouter: React.FC<IAcquisitionRouterProps> = props => {
+  const { path, url } = useRouteMatch();
+
+  if (props.acquisitionFile === undefined || props.acquisitionFile === null) {
+    return null;
+  }
+
+  // render edit forms
+  if (props.isEditing) {
```
Comment (devinleighsmith) [inline source/frontend/src/features/mapSideBar/acquisition/AcquisitionRouter.tsx:40]: what should we do if we are editing and our path doesn't match one of these? currently we display an empty edit screen.

File: source/frontend/src/features/mapSideBar/acquisition/FilePropertyRouter.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  const { path, url } = useRouteMatch();
+
+  if (props.acquisitionFile === undefined || props.acquisitionFile === null) {
+    return null;
+  }
+
+  const fileProperty = getAcquisitionFileProperty(props.acquisitionFile, props.selectedMenuIndex);
+
+  // render edit forms
+  if (props.isEditing) {
```
Comment (devinleighsmith) [inline source/frontend/src/features/mapSideBar/acquisition/FilePropertyRouter.tsx:38]: same question here.

File: source/frontend/src/features/mapSideBar/acquisition/AcquisitionRouter.tsx (hunk 2 above)
Comment (asanchezr) [reply source/frontend/src/features/mapSideBar/acquisition/AcquisitionRouter.tsx:40]: The old code in ViewSelector throws an error - `throw Error('Active edit form not defined');`
I can create a default route that does the same

**Review by devinleighsmith — APPROVED** [review]

File: source/frontend/src/features/mapSideBar/acquisition/FilePropertyRouter.tsx (hunk 3 above)
Comment (asanchezr) [reply source/frontend/src/features/mapSideBar/acquisition/FilePropertyRouter.tsx:38]: I can throw an error like the old code used to do or redirect to file/property details if route doesn't match. Not sure what's better

File: source/frontend/src/features/mapSideBar/acquisition/FilePropertyRouter.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (asanchezr) [reply source/frontend/src/features/mapSideBar/acquisition/FilePropertyRouter.tsx]: Fixed. We show a toast now and also redirect to File Details screen when accessing a non-existing property index.

File: source/frontend/src/features/mapSideBar/acquisition/AcquisitionRouter.tsx (hunk 2 above)
Comment (asanchezr) [reply source/frontend/src/features/mapSideBar/acquisition/AcquisitionRouter.tsx]: Fixed. We redirect to the default File tab (file details)

File: source/frontend/src/features/mapSideBar/acquisition/FilePropertyRouter.tsx (hunk 3 above)
Comment (asanchezr) [reply source/frontend/src/features/mapSideBar/acquisition/FilePropertyRouter.tsx]: Fixed. We redirect to the default property tab (property details)

**Pull request merged**